src/humanloop/eval_utils/run.py

```diff
@@ -0,0 +1,739 @@
+
+            progress_display = f"\r[{bar}] {self._progress}/{self._total}"
+            progress_display += f" ({percentage:.2f}%)"
+
+            if self._progress < self._total:
+                progress_display += f" | ETA: {int(eta)}s"
+            else:
+                progress_display += " | DONE"
+
+            sys.stderr.write(progress_display)
+
+            if self._progress >= self._total:
+                sys.stderr.write("\n")
+
+
+# Module-level so it can be shared by threads.
+_PROGRESS_BAR: Optional[_SimpleProgressBar] = None
+
+
+def run_eval(
+    client: "BaseHumanloop",
+    file: Union[File, Callable],
+    name: Optional[str],
+    dataset: Dataset,
+    evaluators: Optional[Sequence[Evaluator]] = None,
+    workers: int = 4,
+) -> List[EvaluatorCheck]:
+    """
+    Evaluate your function for a given `Dataset` and set of `Evaluators`.
+
+    :param client: the Humanloop API client.
+    :param file: the Humanloop file being evaluated, including a function to run over the dataset.
+    :param name: the name of the Evaluation to run. If it does not exist, a new Evaluation will be created under your File.
+    :param dataset: the dataset to map your function over to produce the outputs required by the Evaluation.
+    :param evaluators: define how judgments are provided for this Evaluation.
+    :param workers: the number of threads to process datapoints using your function concurrently.
+    :return: per Evaluator checks.
+    """
+    global _PROGRESS_BAR
+
+    if isinstance(file, Callable):  # type: ignore
+        # Decorated function
+        file_: File = file.file  # type: ignore
+    else:
+        file_ = file  # type: ignore
+
+    is_decorated = file_.pop("is_decorated", False)
+
+    # Get or create the file on Humanloop
+    version = file_.pop("version", {})
+
+    # Raise error if one of path or id not provided
+    if not file_.get("path") and not file_.get("id"):
+        raise ValueError("You must provide a path or id in your `file`.")
+
+    # Determine the `type` of the `file` to Evaluate - if not `type` provided, default to `flow`
+    try:
+        type_ = typing.cast(FileType, file_.pop("type"))
+        logger.info(
+            f"{CYAN}Evaluating your {type_} function corresponding to `{file_['path']}` on Humanloop{RESET} \n\n"
+        )
+    except KeyError as _:
+        type_ = "flow"
+        logger.warning("No `file` type specified, defaulting to flow.")
+
+    # If a `callable` is provided, Logs will be generated locally, otherwise Logs will be generated on Humanloop.
+    function_ = typing.cast(Optional[Callable], file_.pop("callable", None))
+    if function_ is None:
+        if type_ == "flow":
+            raise ValueError("You must provide a `callable` for your Flow `file` to run a local eval.")
+        else:
+            logger.info(f"No `callable` provided for your {type_} file - will attempt to generate logs on Humanloop.")
+
+    custom_logger = file_.pop("custom_logger", None)
```

Comment:
You are missing my changes:

https://humanloop.slack.com/archives/C024ZHC2LHW/p1730968525286129

Reply:
Assuming this has been resolved given it's outdated, but would prefer you to systematically go through comments and cover them all off as required like discussed @andreibratu